processing/src/test/java/io/druid/segment/filter/BaseFilterTest.java

```diff
@@ -77,14 +77,23 @@
   protected StorageAdapter adapter;
   protected Closeable closeable;
   protected boolean optimize;
+  protected final String testName;
+
+  // JUnit creates a new test instance for every test method call.
+  // For filter tests, the test setup creates a segment.
+  // Creating a new segment for every test method call is pretty slow, so cache the StorageAdapters.
+  // The cache is keyed by "threadID.className", and then "testName"
+  protected static Map<String, Map<String, Pair<StorageAdapter, Closeable>>> adapterCache = new HashMap<>();
```

Comment:
If threadId is part of the key this could be a `ThreadLocal` and not have to worry about synchronizing